page partenaires - créer page #17

Nowhere44 opened the pull request.

Nowhere44 committed:
the page "/partenaires" is done

Vin-CC commented on src/app/(app)/(Partner)/partenaires/_components/partner-search-bar.tsx:
```diff
@@ -0,0 +1,103 @@
+import { Input } from '@/components/ui/input';
+import { PartnerType } from '@prisma/client';
+
+type Partner = {
```
je pense que tu peux utiliser le type issue de prisma plutôt que de le redéclarer

Vin-CC commented on src/app/(app)/(Partner)/partenaires/_components/partner-search-bar.tsx:
```diff
@@ -0,0 +1,103 @@
+                const partners = await response.json();
+                setResults(partners.data);
+            } catch (error) {
+                console.error('Error fetching results:', error);
```
Comme expliquer dans une autre PR, ajoute un message d'erreur.
D'ailleurs est ce qu'il n'y a pas moyen de faire un composant commun ?
C'est quasiment le même à chaque fois, les différences sont quelques labels et la requête de recherche mais sinon la logique de base et le visuel sont les mêmes

Vin-CC commented on src/app/(app)/(Partner)/partenaires/page.tsx:
```diff
@@ -0,0 +1,40 @@
+interface PartnerResponse {
+    data: any[];
+    meta: {
+        limit: number;
```
Pareil que dans d'autres PR, une fois les merges faits, utilises le nouveau Type

Nowhere44 committed:
creating a reusable searchbar

Nowhere44 committed:
many changes have been mada that includes :
-we have changed the way we fetch data in server component
- make searchbar reusable
-usage of revalidatePath instead of Tag
-some pages have been deleted
-changing names of some files and also move them into appropriate folder
